crates/codegen/syntax_templates/src/shared/cst.rs

```diff
@@ -52,4 +53,34 @@ impl Node {
             Self::Token(node) => (&node.text).into(),
         }
     }
+
+    pub fn cursor(&self) -> Cursor {
+        Cursor::new(self.clone())
+    }
+
+    pub fn is_token_kind(&self, kinds: &[TokenKind]) -> Option<&Rc<TokenNode>> {
```

Comment:
For `is_token_kind` and `is_rule_kind`, I wonder why do they take multiple kinds, rather than only one? is it an optimization for NAPI? does that mean that we won't do that if it was a native TS API?

Reply:
That is for uniformity with the Cursor API, where you can search for more than one at a time, without having to do multiple searches over multiple Cursors. In TS it will be `is_token_kind(TokenKind, ...)`. 

Note that it also returns the matched TokenNode, meaning no `match` is required, and it functions therefore as a typeguard.

Note also the following `is_token_matching` which is also for alignment with the cursor API.

Reply:
I think `is_XX()` usually indicates returning booleans, a cheap check that can be chained in a boolean condition with other conditions. Thoughts about something like `as_token()`? which indicates a "cast" that can succeed/fail (thus returning an option).

Reply:
agreed, done